driver-core/src/main/com/mongodb/operation/OperationHelper.java

```diff
@@ -188,26 +221,54 @@ public void call(final AsyncConnection connection, final Throwable t) {
     }
 
     static void validateWriteRequests(final Connection connection, final Boolean bypassDocumentValidation,
-                                        final List<? extends WriteRequest> requests, final WriteConcern writeConcern) {
+                                      final List<? extends WriteRequest> requests, final WriteConcern writeConcern,
+                                      final Boolean retryWrites) {
         checkBypassDocumentValidationIsSupported(connection, bypassDocumentValidation, writeConcern);
         validateWriteRequestCollations(connection, requests, writeConcern);
+        validateRetryWrites(connection, retryWrites);
     }
 
     static void validateWriteRequests(final AsyncConnection connection, final Boolean bypassDocumentValidation,
-                                        final List<? extends WriteRequest> requests, final WriteConcern writeConcern,
-                                        final AsyncCallableWithConnection callable) {
+                                      final List<? extends WriteRequest> requests, final WriteConcern writeConcern,
+                                        final Boolean retryWrites, final AsyncCallableWithConnection callable) {
         checkBypassDocumentValidationIsSupported(connection, bypassDocumentValidation, writeConcern, new AsyncCallableWithConnection() {
             @Override
             public void call(final AsyncConnection connection, final Throwable t) {
                 if (t != null) {
                     callable.call(connection, t);
                 } else {
-                    validateWriteRequestCollations(connection, requests, writeConcern, callable);
+                    validateWriteRequestCollations(connection, requests, writeConcern, new AsyncCallableWithConnection() {
+                        @Override
+                        public void call(final AsyncConnection connection, final Throwable t) {
+                            if (t != null) {
+                                callable.call(connection, t);
+                            } else {
+                                validateRetryWrites(connection, retryWrites, callable);
+                            }
+                        }
+                    });
                 }
             }
         });
     }
 
+    static void validateRetryWrites(final Connection connection, final boolean retryWrites) {
+        if (retryWrites && !serverIsAtLeastVersionThreeDotSix(connection.getDescription())) {
+            throw new IllegalArgumentException(format("Retryable writes are not supported by server version: %s",
+                    connection.getDescription().getServerVersion()));
+        }
+    }
+
+    static void validateRetryWrites(final AsyncConnection connection, final Boolean retryWrites,
```

Comment:
There's three methods in this class where retryWrites is Boolean rather than boolean.  I don't think any are necessary to be nullable.